### PR TITLE
Algo for Two's Complement of any Binary String

### PR DESCRIPTION
**Pull request by atharva20-coder**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by atharva20-coder** [comment]
@siriak @yanglbme Please check the PR

**Comment by atharva20-coder** [comment]
@siriak Please review this PR

File: src/main/java/com/thealgorithms/conversions/TwosComplementOfBinaryString.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	{
+		int n = str.length();
+	
+		// Traverse the string to get first '1' from
+		// the last of string
+		int i;
+		for (i = n-1 ; i >= 0 ; i--)
+			if (str.charAt(i) == '1')
+				break;
+
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/conversions/TwosComplementOfBinaryString.java:16]:
```suggestion
		for (int i = n-1 ; i >= 0 ; i--){
			if (str.charAt(i) == '1'){
				break;
			}
		}
```

File: src/main/java/com/thealgorithms/conversions/TwosComplementOfBinaryString.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		// the last of string
+		for (int i = n-1 ; i >= 0 ; i--){
+			if (str.charAt(i) == '1'){
+				break;
+			}
+		}
+		// If there exists no '1' concat 1 at the
+		// starting of string
+		if (i == -1)
+			return "1" + str;
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/conversions/TwosComplementOfBinaryString.java:20]:
```suggestion
		if (i == -1){
			return "1" + str;
		}
```

File: src/main/java/com/thealgorithms/conversions/TwosComplementOfBinaryString.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	public static void main(String[] args)
+	{
+		Scanner myObj = new Scanner(System.in);  // Create a Scanner object
+    		System.out.println("Enter binary");
+    		String binary = myObj.nextLine();
+		
+		StringBuffer str = new StringBuffer(binary);
+		System.out.println(findTwoscomplement(str));
+	}
+}
```
Comment (debasishbsws) [inline src/main/java/com/thealgorithms/conversions/TwosComplementOfBinaryString.java:48]: Remove main method and ad some jUnit test

**Review by debasishbsws — CHANGES_REQUESTED** [review]

**Pull request closed without merging**